shell/platform/tizen/tizen_vsync_waiter.cc

```diff
@@ -9,43 +9,44 @@
 #include "flutter/shell/platform/tizen/flutter_tizen_engine.h"
 #include "flutter/shell/platform/tizen/logger.h"
 
-static const int kMessageQuit = -1;
-static const int kMessageRequestVblank = 0;
-
 namespace flutter {
 
-typedef struct {
+namespace {
+
+constexpr int kMessageQuit = -1;
+constexpr int kMessageRequestVblank = 0;
+
+struct Message {
   Eina_Thread_Queue_Msg head;
   int event;
   intptr_t baton;
-} Msg;
+};
+
+}  // namespace
 
-TizenVsyncWaiter::TizenVsyncWaiter(FlutterTizenEngine* engine)
-    : engine_(engine) {
-  vblank_thread_ = ecore_thread_feedback_run(RequestVblankLoop, nullptr,
-                                             nullptr, nullptr, this, EINA_TRUE);
+TizenVsyncWaiter::TizenVsyncWaiter(FlutterTizenEngine* engine) {
+  tdm_client_ = std::make_unique<TdmClient>(engine);
+
+  vblank_thread_ = ecore_thread_feedback_run(RunVblankLoop, nullptr, nullptr,
+                                             nullptr, this, EINA_TRUE);
 }
 
 TizenVsyncWaiter::~TizenVsyncWaiter() {
-  if (tdm_client_) {
-    tdm_client_->OnEngineStop();
-  }
-  Send(kMessageQuit, 0);
+  tdm_client_->OnEngineStop();
```

Comment:
How about removing the `OnEngineStop` and doing its job in the `TdmClient`'s destructor?

Reply:
That partially reverts https://github.com/flutter-tizen/engine/pull/204 and seems dangerous (a race condition may occur). Please check again.

Reply:
I think below will be fine...but I think you can leave this as is.

```suggestion
  tdm_client_ = nullptr;
```
```
TdmClient::~TdmClient() {
  {
    std::lock_guard<std::mutex> lock(engine_mutex_);
    engine_ = nullptr;
  }
  DestroyTdm();
}
```

Reply:
Oh, then it makes sense. I'll apply the change.